pkg/tracker/reward/reward.go

```diff
@@ -0,0 +1,418 @@
+			}
+			return
+		}
+		if errC := appender.Commit(); errC != nil {
+			err = errors.Wrap(err, "db append commit failed")
+		}
+	}()
+	lbls := labels.Labels{
+		labels.Label{Name: "__name__", Value: "gas_usage_actual"},
+		labels.Label{Name: "__slot__", Value: slot},
+	}
+
+	_, err = appender.Append(0, lbls, ts, float64(gasUsed))
+	if err != nil {
+		return errors.Wrap(err, "append values to the DB")
+	}
+	return nil
+}
+
+func (self *RewardTracker) recordGasUsageEstimated(ctx context.Context) error {
+	// Getting abi.
+	abi, _ := abi.JSON(strings.NewReader(tellor.TellorABI))
+
+	var (
+		vars struct {
+			Challenge  [32]byte
+			RequestIds [5]*big.Int
+			Difficutly *big.Int
+			Tip        *big.Int
+		}
+		err error
+	)
+
+	// Getting current slot.
+	slot, err := self.Slot()
+	if err != nil {
+		return errors.Wrap(err, "getting current slot")
+	}
+
+	// Getting current challenge.
+	vars, err = self.contractInstance.GetNewCurrentVariables(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return errors.Wrap(err, "call GetNewCurrentVariables")
+	}
+	// We don't have actual values here, so we'll use math.MaxInt64 as our values.
+	// As we tested, this won't make any significant difference in the estimated gas value!
+	reqVals := [5]*big.Int{
+		big.NewInt(math.MaxInt64),
+		big.NewInt(math.MaxInt64),
+		big.NewInt(math.MaxInt64),
+		big.NewInt(math.MaxInt64),
+		big.NewInt(math.MaxInt64),
+	}
+	packed, err := abi.Pack("submitMiningSolution", "", vars.RequestIds, reqVals)
+	if err != nil {
+		return errors.Wrap(err, "packing submitMiningSolution args")
+	}
+	data := ethereum.CallMsg{
+		From: self.addrs[0],
+		To:   &self.contractInstance.Address,
+		// Hardcoded gas price. looks like it doesn't matter in the calculation.
+		GasPrice: big.NewInt(0).Mul(big.NewInt(30), big.NewInt(params.GWei)),
+		Data:     packed,
+	}
+	gasEstimation, err := self.client.EstimateGas(ctx, data)
+	if err != nil {
+		return errors.Wrap(err, "call client.EstimateGas")
+	}
+	err = self.addGasEstimation(slot, gasEstimation)
+	if err != nil {
+		return errors.Wrap(err, "add gas estimation to the db")
+	}
+	return nil
+}
+
+func (self *RewardTracker) addGasEstimation(slot *big.Int, gasEstimation uint64) error {
+	var err error
+	appender := self.tsDB.Appender(self.ctx)
+
+	// Round up the time so that all appends happen with the same TS and
+	// avoid out of order samples errors.
+	ts := timestamp.FromTime(time.Now().Round(5 * time.Second))
+
+	defer func() { // An appender always needs to be committed or rolled back.
+		if err != nil {
+			if err := appender.Rollback(); err != nil {
+				level.Error(self.logger).Log("msg", "db rollback failed", "err", err)
+			}
+			return
+		}
+		if errC := appender.Commit(); errC != nil {
+			err = errors.Wrap(err, "db append commit failed")
+		}
+	}()
+	lbls := labels.Labels{
+		labels.Label{Name: "__name__", Value: "gas_usage_estimated"},
+		labels.Label{Name: "__slot__", Value: slot.String()},
```

Comment:
only the name needs to have underscored, this is prometheus thing
Also rename it in the query
```suggestion
		labels.Label{Name: "slot", Value: slot.String()},
```